app/notifications/rest.py

```diff
@@ -317,62 +285,48 @@ def send_notification(notification_type):
         total_email_count = service_stats.emails_requested
 
         if (total_email_count + total_sms_count >= service.message_limit) and service.restricted:
-            return jsonify(result="error", message='Exceeded send limits ({}) for today'.format(
-                service.message_limit)), 429
+            error = 'Exceeded send limits ({}) for today'.format(service.message_limit)
+            raise InvalidRequest(error, status_code=429)
 
     notification, errors = (
         sms_template_notification_schema if notification_type == 'sms' else email_notification_schema
     ).load(request.get_json())
 
-    if errors:
-        return jsonify(result="error", message=errors), 400
```

Comment:
EmailNotificationSchema and SmsTemplateNotificationSchema are still 'un-strict' so I think you still need this error check